### ISSUE TITLE
Exception raise instead of a simple warning log

### ISSUE THREAD

**Issue by SilviuMilu** [vsajip/python-gnupg#176]
**Requested feature description:**

Due to the fact that instead of raising an error, the code is just logging a warning, I am getting empty files in my workflow.
Example: https://github.com/vsajip/python-gnupg/blob/ba0bb50692a1b32ac78f01faf744b9133120a4ed/gnupg.py#L1071-L1072
My problem is related precisely to these two lines of code, this is the place where I would like to see an exception being raised.

**Desired solution:**

I would like to have an exception being raised, so I can catch it and treat the situation somehow.

**Considered alternates:**

Won't be much. Basically a simple log message won't leave much space for maneuver. I want a way to avoid having just some simple warning when I do have an error.

**Additional context**

![image](https://user-images.githubusercontent.com/97454321/173013515-75189cd5-29f7-461f-b894-5e271bfd563e.png)

**Comment by SwampFalc** [comment]
Are you aware of the fact that all but one operation returns an object which has a `returncode` attribute (https://docs.red-dove.com/python-gnupg/#accessing-gpg-s-return-code)?

To me, that is enough to allow what seems to be your fundamental question, namely detecting and treating the error. For example:

```
result = gpg.encrypt_file(stream, recipients)
if result.returncode != 0:
    fix_error(result)
```
(Sidenote: it is not mentioned in the documentation, but for example `.encrypt_file()` goes even further and adds `.ok` and `.status` attributes to their return value to allow even easier access to and treatment of the error.) 

This is already possible right now, and I do not believe we would gain anything by adding an option to replace this with a try/except block.


(You might notice my own request #177 which I just opened and which is certainly a related issue. However, since I have no way of knowing whether what I wrote above already solves all your problems, or whether you would also want to suppress the printed warnings, I decided to make my request separate so as not to potentially hijack yours.)
